lib/onelogin/ruby-saml/logoutresponse.rb

```diff
@@ -67,46 +98,200 @@ def status_code
         end
       end
 
+      # Gets the StatusMessage from a SAML Response.
+      # @return [String] StatusMessage Value
+      #
+      def status_message
+        @status_message ||= begin
+          node = REXML::XPath.first(document, "/p:LogoutResponse/p:Status/p:StatusMessage", { "p" => PROTOCOL, "a" => ASSERTION })
+          node.text if node
+        end
+      end
+
+      # After execute a validation process, if fails this method returns the causes
+      # @return [Array] Empty Array if no errors, or an Array with the causes
+      #
+      def errors
+        @errors
+      end
+
+      private
+
+      # Gets the expected current_url
+      # (Right now we read this url from the Sinle Logout Service of the Settings)
+      # TODO: Calculate the real current_url and use it.
+      # @return 
+      #
+      def current_url
+        @current_url ||= begin
+          unless self.settings.nil? or self.settings.single_logout_service_url.nil?
+            self.settings.single_logout_service_url
+          end
+        end
+      end
+
+      # Validates the Logout Response (calls several validation methods)
+      # If fails, the attribute errors will contains the reason for the invalidation.
+      # @param [Boolean] soft Enable or Disable the soft mode (In order to raise exceptions when the logout response is invalid or not)
+      # @param [String|nil] request_id The ID of the Logout Request sent by this SP to the IdP (if was sent any)
+      # @return [Boolean|ValidationError] True if the Logout Response is valid, otherwise
```

Comment:
```ruby
# Validates the Logout Response (calls several validation methods)
# If fails, the attribute errors will contains the reason for the invalidation.
# @param [Boolean] soft Enable or Disable the soft mode in order to raise exceptions if logout response is invalid
# @param [String|nil] request_id The ID of the Logout Request sent by this SP to the IdP (if was sent any)
# @return [Boolean|ValidationError] true if the Logout Response is valid, otherwise false if soft == true
# @raise [ValidationError] if soft == false
```